### PR TITLE
Improved Sec Starting Gear

### PR DESCRIPTION
**Pull request by Surrealaser**
## About The Pull Request

Per title.

HoS, Sec Officers, Wardens and Deputies all start with a basic security belt loaded with a stun baton, pepper spray, flash, sec light, and sec radio (non-cuff duplicate gear has been removed). The PDA has been moved to a pocket slot.

HoS, Sec Officers and Wardens have an 'improved' starter box that has 3 glow sticks and a crowbar, in the event of late joining in a depowered environment.

Sec Officers now start with SecHUD glasses.

Deputies now start with a basic SecHUD scouter for determining the power level of enemy criminals.

Security radio item added as standard gear in the security belt; this radio can access the full range of frequencies, has optimal subspace transmission and starts at a randomized frequency out of standard station bounced range, allowing for emergency communication without relating information to everyone. A user can revert to this randomized security frequency by swiping the radio with a Sec access ID.

Most security lockers have had their contents deleted as a result of these changes, and gear being moved to security vendors, as well as the addition of new security protolathe designs (handcuffs, stun batons, sec armor, clothing and sec radios).

## Why It's Good For The Game

Allows late joining security to always be properly equipped to do their job, and deal with depowered arrivals; lessens dependence on a locker room that may be looted or dry of belts and sec glasses by the time the officer arrives. 

Allows for subsequent PRs to remove 'locker bloat' from maps.

## Changelog
:cl:
add: Wardens, Sec Officers and Deputies have more of their standard loadout on spawning, making these roles more late joiner friendly.
add: Basic security protolathe designs added (handcuffs, stun batons, sec armor, clothing and sec radios).
add: Security station bounced radio added. Can be reset to its default frequency by swiping with a Security access ID.
tweak: Security locker gear either removed or moved to security vendor in light of these changes.
/:cl:

### CODE REVIEW

**Comment by GuillaumePrata** [comment]
Removing 'locker bloat' is good, this is a good first step.

File: code/FulpstationCode/deputy.dm [hunk 1]
```diff
@@ -86,15 +86,15 @@
...
+	belt = /obj/item/storage/belt/security/fulp_starter_full
 	ears = /obj/item/radio/headset/headset_sec
 	uniform = /obj/item/clothing/under/rank/security/mallcop
 	accessory = /obj/item/clothing/accessory/armband/deputy
 	shoes = /obj/item/clothing/shoes/jackboots
 	l_pocket = /obj/item/restraints/handcuffs/cable/zipties
-	r_pocket = /obj/item/assembly/flash/handheld
-	glasses = /obj/item/clothing/glasses/hud/security/sunglasses
-	backpack_contents = list(/obj/item/melee/baton/loaded=1)
+	r_pocket = /obj/item/pda/security
```
Comment (CosmicPhoenix) [inline code/FulpstationCode/deputy.dm:95]: If a PDA isn't in the belt slot you have to direct the game to the new PDA using "pda_slot = "
In this case pda_slot = SLOT_R_STORE works.

File: code/FulpstationCode/fulp_starter_gear/fulp_starter_gear.dm [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+	new /obj/item/assembly/flash/handheld(src)
+	new /obj/item/melee/baton/loaded(src)
+	update_icon()
+
+/obj/item/storage/box/security/improved/PopulateContents()
+	..() // we want the regular stuff too
+	new /obj/item/crowbar/red(src)
+	new /obj/item/flashlight/glowstick(src)
+	new /obj/item/flashlight/glowstick(src)
+	new /obj/item/flashlight/glowstick(src)
```
Comment (CosmicPhoenix) [inline code/FulpstationCode/fulp_starter_gear/fulp_starter_gear.dm]: Adding the glowsticks I feel is kind of redundant since they're already spawning with a Seclight. Assuming #168 is merged as well this would give them the civilian glowsticks rather then the military ones; it makes the light pouch fairly redundant too I feel.

**Review by CosmicPhoenix — CHANGES_REQUESTED** [review]
Suggested a couple changes to be made. Main issue being the PDAs now working on spawn.

**Comment by CosmicPhoenix** [comment]
Alright, so, I personally don't think Sec should be given belts; the crowbar is fine however. Giving loaded belts to Security makes their lockers pretty pointless, they don't need anything out of them so they're just going to sit in the locker room untouched until someone breaks into it. 

Since you're already giving a crowbar to Sec they should be able to get into the Brig without much trouble allowing them to equip themselves there with the belt.

As for the locker bloat it isn't a huge problem for Security, the lockers only have 7 - 8 items. Removing the standard headset from the lockers wouldn't be a bad idea though as they're completely redundant except for providing an extra encryption key.

File: code/FulpstationCode/fulp_starter_gear/fulp_starter_gear.dm [hunk 3]
```diff
@@ -0,0 +1,14 @@
+/obj/item/storage/belt/security/fulp_starter_full/PopulateContents()
+	new /obj/item/reagent_containers/spray/pepper(src)
+	new /obj/item/flashlight/seclite(src)
+	new /obj/item/radio(src)
+	new /obj/item/assembly/flash/handheld(src)
+	new /obj/item/melee/baton/loaded(src)
+	update_icon()
+
+/obj/item/storage/box/security/improved/PopulateContents()
+	..() // we want the regular stuff too
```
Comment (CosmicPhoenix) [inline code/FulpstationCode/fulp_starter_gear/fulp_starter_gear.dm]: On a complete side note, using ..() in objects to inherit inventories is a really good idea, I'll absolutely be trying this in #168 to see if I can make the vendor overwrites smaller.

**Comment by ghost** [comment]
Damn cosmic, I'll leave this one with you

**Comment by Surrealaser** [comment]
> Alright, so, I personally don't think Sec should be given belts; the crowbar is fine however. Giving loaded belts to Security makes their lockers pretty pointless, they don't need anything out of them so they're just going to sit in the locker room untouched until someone breaks into it.
> 
> Since you're already giving a crowbar to Sec they should be able to get into the Brig without much trouble allowing them to equip themselves there with the belt.
> 
> As for the locker bloat it isn't a huge problem for Security, the lockers only have 7 - 8 items. Removing the standard headset from the lockers wouldn't be a bad idea though as they're completely redundant except for providing an extra encryption key.

Ultimately, I feel it's for the best if we work towards phasing out the need for the lockers, and maybe have them feature spare uniforms, glasses and headsets and such in the event you need to replace clothing due to fire, deputize other crew in the event of nukies/blob/cult/etc, or to simply use the locker room as secure storage while relying more heavily on secure vendors. Sec in general relies on accessing their content, the utility belt especially, far too much to do their job, and they're screwed if they can't, especially if they're latejoiners stepping into the heart of chaos. Even fully geared, it's probably the hardest role in the game, nevermind when you're lacking key tools from your kit.

File: code/FulpstationCode/fulp_starter_gear/fulp_starter_gear.dm (hunk 2 above)
Comment (Surrealaser) [reply code/FulpstationCode/fulp_starter_gear/fulp_starter_gear.dm]: I have no strong opinions on this one; I could remove the glowsticks.

File: code/FulpstationCode/deputy.dm (hunk 1 above)
Comment (Surrealaser) [reply code/FulpstationCode/deputy.dm:95]: You do not; I tested this.

File: code/FulpstationCode/fulp_starter_gear/fulp_starter_gear.dm (hunk 2 above)
Comment (Surrealaser) [reply code/FulpstationCode/fulp_starter_gear/fulp_starter_gear.dm]: Also, during testing I learned something interesting: you can't simply null out backpack contents to properly override; you must utilize an empty list instead.

File: code/FulpstationCode/deputy.dm (hunk 1 above)
Comment (CosmicPhoenix) [reply code/FulpstationCode/deputy.dm:95]: It was a big problem with adding belts to EMTs and Brig Docs on spawn so I don't see why it'd suddenly work here. The PDA isn't linked to the player on spawn if you don't change the pda_slot, it just spawns a userless PDA that requires the player to manually link the PDA to them using their ID Card which is somewhat annoying to do every round.

**Comment by CosmicPhoenix** [comment]
> Ultimately, I feel it's for the best if we work towards phasing out the need for the lockers, and maybe have them feature spare uniforms, glasses and headsets and such in the event you need to replace clothing due to fire, deputize other crew in the event of nukies/blob/cult/etc, or to simply use the locker room as secure storage while relying more heavily on secure vendors.

I agree with relying more heavily on vendors. Personally I'd prefer if the belts spawned empty and you had to "buy" the equipment from the vendor, that way you don't snag items from other belts to load up on certain items. The flash for instance I always drop in favor of another pair of handcuffs.

> Sec in general relies on accessing their content, the utility belt especially, far too much to do their job, and they're screwed if they can't, especially if they're latejoiners stepping into the heart of chaos. Even fully geared, it's probably the hardest role in the game, nevermind when you're lacking key tools from your kit.

I think that generally the belt is an unnecessary add-on to Security Officer kits. The flash and flashbang are almost never used, and the pepper spray is borderline useless normally. The only important items inside the belt are the handcuffs and extra baton, both of which Security Officers already spawn with.

If you _really_ want to spawn Security Officers with belts I'd go with giving them bare minimum equipment inside the belt and have them travel to the Brig to get more. I'd move belts out of the lockers too and place it into the Sec ClothesMate or SecVend machines.

Also, I don't think Deputies should get the belt at all. It's supposed to be a beginner's job for Sec so I don't think they should be over loaded with equipment. Deputies having the bare necessities let's people become competent with the couple items they start with. Thematically too Deputies are just glorified mall cops, they don't even start with handcuffs they just have zip-ties.

**Comment by Surrealaser** [comment]
@CosmicPhoenix Everything in the sec officer's kit is effective; it's really a matter of using the right tool in the right situation. I don't agree with discounting the flash (which is a win button against anyone without protection, and crucial vs malf/nuke borgs) and the pepper spray (which is highly and instantly effective against most targets, even if they have flash protection). Also the flashbang is extremely powerful, and can turn an otherwise impossible to approach antag into a cakewalk; it's hard to overestimate it, which is  why I was careful to specifically exclude it from the spawn belt gear.

Sure, I could scale back the gear for the deputy; really the only thing he possibly shouldn't have in the belt is the pepper spray though; recall that most of the belt inventory are simply items that the role normally gets anyways, but which has been instead placed inside the belt.

File: code/FulpstationCode/deputy.dm (hunk 1 above)
Comment (Surrealaser) [reply code/FulpstationCode/deputy.dm:95]: I see, so you're saying the problem is that it isn't autolinked which creates a minor inconvienience of having to manually link via your ID. Sure, I can fix that.

**Comment by Surrealaser** [comment]
Security radio item added as standard gear in the security belt; this radio can access the full range of frequencies, has optimal subspace transmission and starts at a randomized frequency out of standard station bounced range, allowing for emergency communication without relaying information to everyone.

Most security lockers now have their contents deleted as a result of these changes, and gear being moved to security vendors, as well as the addition of new basic security protolathe designs (handcuffs, stun batons, sec armor, clothing and sec radios).
